src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -114,12 +119,20 @@ public function process(ContainerBuilder $container)
             }
             $class = $r->name;
 
+            $dispatcherDefinition = $globalDispatcherDefinition;
+            foreach ($tags as $attributes) {
+                if (isset($attributes['dispatcher'])) {
+                    $dispatcherDefinition = $container->getDefinition($attributes['dispatcher']);
+                    break;
```

Comment:
Not sure about this logic in the foreach: this creates a correlation between tags, while they should be independent to me. Otherwise, we don't allow one listener to subscribe to several dispatchers, which looks like a serious limitation to me.

Reply:
Please note that this is for subscribers only, listeners are done before this code.

The current behavior didn't loop over all subscriber tags, only registering an event subscriber once. I figured it was best to keep that behavior. However, I see your point about being able to register on multiple dispatchers, I'll implement it that way 👍

Reply:
> Please note that this is for subscribers only, listeners are done before this code.

we should make it work for listeners also, isn't it?

Reply:
Lines 93 - 98 are for listeners. Listeners have a `method` and `event` attribute, so it already iterates over every tag (see line 67).

Reply:
But they don't support a `dispatcher` attribute, do they?

Reply:
Yes they do :) This is the whole diff, first block adds `dispatcher` for listeners, second block adds `dispatcher` for event subscribers:
![](https://user-images.githubusercontent.com/749025/78071935-93ce0180-739e-11ea-82c5-44159b5d31f6.png)

Reply:
OK got it sorry for the noise :)

Reply:
No problem!